### PR TITLE
Update GitHub Action and .NET related content to include workload and SDK callouts

### LINKED ISSUE #26368: Add details about `dotnet workload install` to existing GitHub Actions content
We need to add callouts about [`dotnet workload install`](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-workload-install) to the existing GitHub Actions content in the .NET docs, including but not limited to:

- [Create a publish app GitHub Action](https://docs.microsoft.com/en-us/dotnet/devops/dotnet-publish-github-action)
- [DevOps for ASP.NET Core Developers](https://docs.microsoft.com/en-us/dotnet/architecture/devops-for-aspnet-developers)

/cc @timheuer and @KathleenDollard

### PR DESCRIPTION
**Pull request by IEvangelist**
## Summary

- Added various details about the .NET SDK being a requirement of .NET-based GitHub Action workflows.
- Add examples related to `dotnet workload install` as it relates to GitHub Actions.
- Minor editorial updates, clean up and fixes.

Fixes #26368

## Internal preview links

✔️ [Build a .NET web app using GitHub Actions: Workflow structure](https://review.docs.microsoft.com/en-us/dotnet/architecture/devops-for-aspnet-developers/actions-build?branch=pr-en-us-26386#workflow-structure)
✔️ [GitHub Actions and .NET](https://review.docs.microsoft.com/en-us/dotnet/devops/github-actions-overview?branch=pr-en-us-26386)
✔️ [Create a build validation GitHub Action: Create a workflow file](https://review.docs.microsoft.com/en-us/dotnet/devops/dotnet-build-github-action?branch=pr-en-us-26386#create-a-workflow-file)

/cc @timheuer and @KathleenDollard

### CODE REVIEW

File: docs/architecture/devops-for-aspnet-developers/actions-build.md [hunk 1]
```diff
@@ -35,6 +35,18 @@ The `steps` node can be as easy as inline commands, or they can be actions. Most
 > [!TIP]
 > For more information, see [GitHub Actions YAML syntax](https://docs.github.com/actions/reference/workflow-syntax-for-github-actions).
 
+From a workflow file, you're able to `run` any of the available [.NET CLI commands](../../core/tools/index.md). For example, if you're required to build, test, and deploy an ASP.NET Core Blazor WebAssembly app with Ahead-of-Time (AoT) compilation, you'd use the following commands:
```
Comment (timheuer) [inline docs/architecture/devops-for-aspnet-developers/actions-build.md:38]: I don't know if we have docs on Wasm AOT but wonder if we should link to that here?

File: docs/devops/dotnet-build-github-action.md [hunk 2]
```diff
@@ -43,10 +43,22 @@ In the preceding workflow composition:
   :::code language="yml" source="snippets/dotnet-build-github-action/build-validation.yml" range="14-34" highlight="2,4-8,13-15,18,21":::
 
   - There is a single job, named `build-<os>` where the `<os>` is the operating system name from the `strategy/matrix`. The `name` and `runs-on` elements are dynamic for each value in the `matrix/os`. This will run on the latest versions of Ubuntu, Windows, and macOS.
-  - The `actions/setup-dotnet@v1` GitHub Action is used to setup the .NET SDK with the specified version from the `DOTNET_VERSION` environment variable.
+  - The `actions/setup-dotnet@v1` GitHub Action is required to set up the .NET SDK with the specified version from the `DOTNET_VERSION` environment variable.
+  - (Optionally) Additional steps may be required, depending on your .NET workload. They're omitted from this example, but you may need additional tools installed to build your apps.
+    - For example, when building an ASP.NET Core Blazor WebAssembly application with Ahead-of-Time (AoT) compilation you'd install the corresponding workload.
```
Comment (timheuer) [inline docs/devops/dotnet-build-github-action.md:48]: "you'd install the corresponding workload before running restore/build/publish operations"

**Review by timheuer — DISMISSED** [review]

File: docs/architecture/devops-for-aspnet-developers/actions-build.md (hunk 1 above)
Comment (IEvangelist) [reply docs/architecture/devops-for-aspnet-developers/actions-build.md:38]: I'll check, if we have them - I'll add them.

File: docs/devops/dotnet-build-github-action.md (hunk 2 above)
Comment (IEvangelist) [reply docs/devops/dotnet-build-github-action.md:48]:
```suggestion
    - For example, when building an ASP.NET Core Blazor WebAssembly application with Ahead-of-Time (AoT) compilation you'd install the corresponding workload before running restore/build/publish operations.
```

File: docs/devops/github-actions-overview.md [hunk 3]
```diff
@@ -21,7 +21,7 @@ GitHub Actions represent standalone commands, such as:
 
 While these commands are isolated to a single action, they're powerful through *workflow composition*. In workflow composition, you define the *events* that trigger the workflow. Once a workflow is running, there are various *jobs* it's instructed to perform &mdash; with each job defining any number of *steps*. The *steps* delegate out to GitHub Actions, or alternatively call command-line scripts.
 
-For more information, see [Introduction to GitHub Actions](https://docs.github.com/actions/learn-github-actions/introduction-to-github-actions).
+For more information, see [Introduction to GitHub Actions](https://docs.github.com/actions/learn-github-actions/introduction-to-github-actions). You should think of a workflow file as your composition that represents the various steps to either build, test, and/or publish your application. There are many [.NET CLI commands](../core/tools/index.md) available, most of which could be used in the context of a GitHub Action.
```
Comment (tdykstra) [inline docs/devops/github-actions-overview.md:24]:
```suggestion
For more information, see [Introduction to GitHub Actions](https://docs.github.com/actions/learn-github-actions/introduction-to-github-actions). Think of a workflow file as a composition that represents the various steps to build, test, and/or publish an application. Many [.NET CLI commands](../core/tools/index.md) are available, most of which could be used in the context of a GitHub Action.
```

File: docs/devops/github-actions-overview.md [hunk 4]
```diff
@@ -84,7 +84,7 @@ To use *encrypted secrets* in your workflow files, you reference the secrets usi
 ${{ secrets.MY_SECRET_VALUE }} # The MY_SECRET_VALUE must exist in the repository as a secret
 ```
 
-Secret values are never printed to the logs, instead their names are printed with asterisk representing their values. For example, as each step runs within a job &mdash; all of the values it uses are output to the action log. When secret values are out, they render similar to the following:
+Secret values are never printed in the logs, instead, their names are printed with the asterisk representing their values. For example, as each step runs within a job &mdash; all of the values it uses are output to the action log. When secret values are out, they render similar to the following:
```
Comment (tdykstra) [inline docs/devops/github-actions-overview.md:87]:
```suggestion
Secret values are never printed in the logs. Instead, their names are printed with an asterisk representing their values. For example, as each step runs within a job, all of the values it uses are output to the action log. Secret values render similar to the following:
```

**Review by tdykstra — APPROVED** [review]
LGTM, a couple minor suggestions.

**Pull request merged**